packages/x-components/src/components/animations/__tests__/animations-abstract-factory.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+import { getDataTestSelector } from '../../../__tests__/utils';
+import { animationAbstractFactory } from '../animation.abstract-factory';
+
+describe('testing animation abstract factory', () => {
+  function renderAnimatedComponent(animationName: string): RenderApi {
+    const wrapper = mount(
+      {
+        template: `
+          <animationComponent>
+            <div v-if="open" data-test="animation-content"></div>
+          </animationComponent>
+        `,
+        components: {
+          animationComponent: animationAbstractFactory(animationName)()
+        },
+        props: {
+          open: { default: false }
+        }
+      },
+      {}
+    );
+
+    return {
+      toggleOpen(): void | Promise<void> {
+        return wrapper.setProps({ open: true });
+      },
+      getContentWrapper(): Wrapper<Vue> {
+        return wrapper.find(getDataTestSelector('animation-content'));
+      },
+      wrapper
+    };
+  }
+
+  // eslint-disable-next-line max-len
+  it('renders the animation component created by abstract factory.', async () => {
```

Comment:
I agree. Instead of checking if the wrapper exists, I think we should be checking that a `transition` node is being rendered with a prop `name` containing ``x-${animationName}--${animationOrigin} x-${animationName}-``